### PR TITLE
Measure cleanup

### PR DESCRIPTION
**Pull request by anstepp**
All tests pass, measure and part rewritten for cleaner, more simplified, and more maintainable logic.

### CODE REVIEW

File: py2musicxml/notation/beat.py [hunk 1]
```diff
@@ -16,21 +14,21 @@ def __init__(self, subdivisions: int) -> None:
...
     def add_note(self, note: Note) -> None:
         logging.debug(f"Appending note: {note}")
         self.notes.append(note)
-        self._make_beams()
 
     def extend_beat(self, notes: Iterable[Note]) -> None:
         self.notes.extend(notes)
-        self._make_beams()
 
     #     self._tuplet_test()
```
Comment (rachhouse) [inline py2musicxml/notation/beat.py:33]: Can this be deleted?

File: py2musicxml/notation/chord.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        for idx, note in enumerate(note_list):
+            log.debug(f"Note list, idx: {idx}, note {note}")
+
+        self.notes = self._sort_notes(note_list)
+        self.dur = self.notes[0].dur
+
+        for note in self.notes[1:]:
+            note.is_chord_member = True
+
+    def _sort_notes(self, note_list: Iterable[Note]) -> Iterable[Note]:
```
Comment (rachhouse) [inline py2musicxml/notation/chord.py]: While you can accept an `Iterable` if you're just doing a `for x in y` pattern, it's best to return type hint `List` if that's what you are actually returning.

File: py2musicxml/notation/chord.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            if new_duration > 0:
+                self.dur = new_duration
+                for note in self.notes:
+                    note.dur = new_duration
+        except ValueError as e:
+            logging.error(e)
+            raise
+
+
+    def split(self, diff) -> Tuple["__class__", "__class__"]:
```
Comment (rachhouse) [inline py2musicxml/notation/chord.py]: Do you still have to do this:
```
Tuple["__class__", "__class__"]
```
Or is there a newer Python construct?

**Review by rachhouse — COMMENTED** [review]
It might be helpful for future you and non-music developers to have more comments on the tests about what they are testing and why.

File: py2musicxml/notation/beat.py (hunk 1 above)
Comment (anstepp) [reply py2musicxml/notation/beat.py:33]: Yes! I had deleted it after the push. Oops.

**Review by rachhouse — APPROVED** [review]

**Pull request merged**